src/doc/rustc/src/platform-support/wasm32-unknown-emscripten.md

```diff
@@ -0,0 +1,157 @@
+# `wasm32-unknown-emscipten`
+
+**Tier: 2**
+
+The `wasm32-unknown-emscripten` target is a WebAssembly compilation target which
+uses the [Emscripten](https://emscripten.org/) compiler toolchain. Emscripten is
+a C/C++ toolchain designed to make it as easy as possible to port C/C++ code
+written for Linux to run on the web. It thus provides POSIX-compatible (musl) `libc`
+and `libstd` implementations and many Linux APIs, access to the OpenGL and SDL APIs,
+and the ability to run arbitrary JavaScript code, all based on web APIs using JS
+glue code. With the `wasm32-unknown-emscripten` target, Rust code can interoperate
+with Emscripten's ecosystem, C/C++ and JS code, and web APIs.
+
+One existing user of this target is the
+[`pyodide` project](https://pyodide.org/) which provides a Python runtime in
+WebAssembly using Emscripten and compiles Python extension modules written in Rust
+to the `wasm32-unknown-emscripten` target.
+
+If you want to generate a standalone bare-bones WebAssembly binary that does not
+require JS glue code and access to the web APIs and can thus run outside the web,
+the [`wasm32-unknown-unknown`](./wasm32-unknown-unknown.md) target may be better
+suited for you. However, [`wasm32-unknown-unknown`](./wasm32-unknown-unknown.md)
+only has partial support for the Rust standard library and does not (easily)
+support interop with C/C++ code.
+
+Like Emscripten, the WASI targets [`wasm32-wasip1`](./wasm32-wasip1.md) and
+[`wasm32-wasip2`](./wasm32-wasip2.md) also provide access to the host environment,
+support interop with C/C++ (and other languages), and support most of the Rust
+standard library. While the WASI targets are portable across different hosts
+(web and non-web), WASI has no standard way of accessing web APIs, whereas
+Emscripten has the ability to run arbitrary JS from WASM and access many web APIs.
+If you are only targeting the web and need to access web APIs, the
+`wasm32-unknown-emscripten` target may be preferable.
+
+## Target maintainers
+
+- Hood Chatham, https://github.com/hoodmane
+- Sam Clegg, https://github.com/sbc100
+- Juniper Tyree, https://github.com/juntyr
+
+## Requirements
+
+This target is cross-compiled. The Emscripten compiler toolchain (emsdk) must be
+installed to link WASM binaries with `emcc`. Please refer to
+<https://emscripten.org/docs/getting_started/downloads.html> on how to get started
+with the emsdk.
+
+You can also build the emsdk from source with:
```

Comment:
This does not build from source. emsdk is just an installer for the toolchain.

Interestingly, [the code for the emscripten compiler toolchain](https://github.com/emscripten-core/emscripten/) itself has no compilation step, it is implemented in a mixture of Python and JavaScript. So you can just modify the source files in place and immediately have a patched compiler. The standard library / sysroot is compiled as needed (since `emcc` implements many many different ABIs it doesn't make sense to ship prebuilt static libraries, it builds them as needed and caches them).

All of the compiled code needed for implementing the compiler (as opposed to host wasm code) is upstream either in [binaryen](https://github.com/WebAssembly/binaryen) or in llvm itself.

Reply:
Thanks for correcting my mistake - I've reworded the paragraph